### PR TITLE
Add enet example

### PR DESCRIPTION
**Pull request by QB3**
The goal is to add an example with enet with 2 hyperparameters

### CODE REVIEW

**Comment by QB3** [comment]
Good to go for me

File: examples/plot_held_out_enet.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
+"""
+============================
+Lasso with held-out test set
```
Comment (agramfort) [inline examples/plot_held_out_enet.py:3]: Elastic-Net

File: examples/plot_held_out_enet.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    y = X @ beta + np.random.randn(X.shape[0])
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33)
+    X_train, X_val, y_train, y_val = train_test_split(
+        X_train, y_train, test_size=0.5)
+# X_train, X_val, X_test, y_train, y_val, y_test = get_data(dataset)
+# if use_small_part:
+#     idx = np.abs((X_train.T @ y_train)).argsort()[-1000:]
+#     X_train = X_train[:, idx]
+#     X_val = X_val[:, idx]
+#     X_test = X_test[:, idx]
```
Comment (agramfort) [inline examples/plot_held_out_enet.py:56]: cleanup

File: examples/plot_held_out_enet.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+print("Finished loading data")
+
+alpha_max = np.max(np.abs(X_train.T @ y_train))
```
Comment (agramfort) [inline examples/plot_held_out_enet.py:60]: for nicer rendering of examples with sections use a line like this:

############################################################

File: examples/plot_held_out_enet.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            alpha=(alphas_1[i] + alphas_2[j]), fit_intercept=False,
+            l1_ratio=alphas_1[i] / (alphas_1[i] + alphas_2[j]),
+            tol=tol, max_iter=max_iter, warm_start=False)
+        clf.fit(X_train, y_train)
+        results[i, j] = norm(y_val - X_val @ clf.coef_) ** 2 / X_val.shape[0]
+t_grid_search += time.time()
+print("Finished grid-search")
+
+
+# grad search
```
Comment (agramfort) [inline examples/plot_held_out_enet.py:93]: insert section here

File: examples/plot_held_out_enet.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+alphas_grad = np.exp(np.array(monitor.log_alphas))
+alphas_grad /= alpha_max
+
+
+print("Time grid-search %f" % t_grid_search)
+print("Time grad-search %f" % t_grad_search)
+print("Minimum grid search %0.3e" % results.min())
+print("Minimum grad search %0.3e" % np.array(monitor.objs).min())
+
+# plot results
```
Comment (agramfort) [inline examples/plot_held_out_enet.py:119]: insert section here

**Pull request merged**